harness/features/allVariables.cloud.test.ts

```diff
@@ -37,6 +38,19 @@ describe('allVariables Tests - Cloud', () => {
                 expect(variablesMap).toMatchObject({})
             })
 
+            it('should throw if variables request fails on user error', async () => {
+                scope
+                    .post(`/client/${client.clientId}/v1/variables`)
+                    .reply(401, { message: 'Invalid sdk token' })
+
+                const response = await client.callAllVariables({
+                    user_id: 'test_user',
+                    email: 'user@gmail.com'
+                }, true)
+                const res = await response.json()
+                expectErrorMessageToBe(res, 'Invalid sdk token')
```

Comment:
I think youll need to more specifically pass in `res.asyncError`

Reply:
good catch, fixed!